ui/src/app/pages/data/data-component.tsx

```diff
@@ -78,202 +77,175 @@ const dataSetImg = '/assets/images/dataset-diagram.svg';
 
 const resourceTypesToFetch = [ResourceType.COHORT, ResourceType.COHORTREVIEW, ResourceType.CONCEPTSET, ResourceType.DATASET];
 
-export const DataComponent = withCurrentWorkspace()(class extends React.Component<
-  {workspace: WorkspaceData},
-  {activeTab: Tabs, resourceList: WorkspaceResource[], isLoading: boolean,
-    creatingConceptSet: boolean, existingDataSetName: string[], existingCohortName: string[],
-    existingCohortReviewName: string[], existingConceptSetName: string[]}> {
+export const DataComponent = withCurrentWorkspace()(({workspace}) => {
```

Comment:
Arguably this change could be split out to reduce risk - unless this has some dependency on the other changes that I'm missing.

Reply:
There's a functional change here which fixes a bug in the previous implementation. Previously, changes to the workspace prop would rerender the component but _not_ reload new resources since that call is only made on `componentDidMount`.

Net result was that copying a resource to a different workspace would not show the resources for the new workspace, rather the old one. (Side note, I would have thought that angular would force the whole component to rerender but I guess that wasn't the case.)

I was working on the fix and thought that `useEffect()` was a good solution so I converted the component to hooks while I was here.

Ideally, I could have shipped this out first but ended up bundling it b/c of the pre vacation crunch.